api/global/internal/trace.go

```diff
@@ -0,0 +1,78 @@
+package internal
+
+import (
+	"context"
+	"sync"
+
+	"go.opentelemetry.io/otel/api/trace"
+)
+
+type traceProvider struct {
+	lock    sync.Mutex
+	tracers []*tracer
+
+	delegate trace.Provider
+}
+
+var _ trace.Provider = &traceProvider{}
```

Comment:
I hate to ask this, but... please add a comment here to make clear that this is an interface satisfaction declaration to prevent silent test failures. e.g. to prevent people from needing to ask  https://stackoverflow.com/questions/30367803/why-declare-like-var-i-t-and-var-i-t-continuously/30367931#30367931

Reply:
Sure thing 👍